pyiceberg/table/update/validate.py

```diff
@@ -0,0 +1,70 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from pyiceberg.manifest import ManifestContent, ManifestFile
+from pyiceberg.table import Table
+from pyiceberg.table.snapshots import Operation, Snapshot, ancestors_between
+
+
+class ValidationException(Exception):
+    """Raised when validation fails."""
+
+
+def validation_history(
+    table: Table,
+    starting_snapshot: Snapshot,
```

Comment:
`starting_snapshot` should be Optional, because it could be missing. If the `starting_snapshot` for validation is missing, and if the isolation level is specified (not None) then we'd need to run the validation check from the beginning of the table's history - this is the current behavior in the Java reference implementation

Reply:
Ok I think I see, I think the naming is a little confusing and it makes things easier to think about if we stick to `from_snapshot` and `to_snapshot`. Just pushed some changes, let me know what you think!

Reply:
Thanks @jayceslesar ! Unfortunately I now find this more confusing, because the order of `from_snapshot` and `to_snapshot` are different in the `ancestor_between` function header from the function call within `validation_history`


```
def ancestors_between(
    to_snapshot: Snapshot, from_snapshot: Optional[Snapshot], table_metadata: TableMetadata
) -> Iterable[Snapshot]:
```

The function call in `validation_history`:
```
    for snapshot in ancestors_between(from_snapshot, to_snapshot, table.metadata):
```

Is this intended?

Reply:
im gonna rock with from, to in that order -- should be done